### PR TITLE
LPS-89235 Create headless content-space api and impl to ease development of generator

### PR DESCRIPTION
**Pull request by ruben-pulido**

### CODE REVIEW

File: modules/apps/headless/content-space/content-space-headless-apio-api/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Bundle-Name: Liferay Content Space Headless Apio API
+Bundle-SymbolicName: com.liferay.content.space.headless.apio.api
+Bundle-Version: 2.0.0
```
Comment (nhpatt) [inline modules/apps/headless/content-space/content-space-headless-apio-api/bnd.bnd]: Should be 1

File: modules/apps/headless/content-space/content-space-headless-apio-api/bnd.bnd [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Bundle-Name: Liferay Content Space Headless Apio API
```
Comment (nhpatt) [inline modules/apps/headless/content-space/content-space-headless-apio-api/bnd.bnd]: remove apio, from folder and bundle names. Should be just content-space (or what brian asked headless-content-space)

File: modules/apps/headless/content-space/content-space-headless-apio-api/build.gradle [hunk 3]
```diff
@@ -0,0 +1,6 @@
+dependencies {
+	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "default"
+	compileOnly project(":apps:apio-architect:apio-architect-api")
+	compileOnly project(":apps:headless-apio:folder:folder-apio-api")
```
Comment (nhpatt) [inline modules/apps/headless/content-space/content-space-headless-apio-api/build.gradle]: this is going to be fun.

File: modules/apps/headless/content-space/content-space-headless-apio-api/src/main/java/com/liferay/content/space/headless/apio/architect/identifier/ContentSpaceIdentifier.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * Holds information about a content space identifier. The internal method
+ * {@code com.liferay.portal.kernel.model.GroupModel#getGroupId()} returns the
+ * identifier.
+ *
+ * @author     Javier Gamarra
+ * @deprecated As of Mueller (7.2.x)
+ */
+@Deprecated
+public interface ContentSpaceIdentifier extends Identifier<Long> {
```
Comment (nhpatt) [inline modules/apps/headless/content-space/content-space-headless-apio-api/src/main/java/com/liferay/content/space/headless/apio/architect/identifier/ContentSpaceIdentifier.java]: mmm. Not needed right?

File: modules/apps/headless/content-space/content-space-headless-apio-impl/bnd.bnd [hunk 5]
```diff
@@ -0,0 +1,4 @@
+Bundle-Name: Liferay Content Space Headless Apio Implementation
+Bundle-SymbolicName: com.liferay.content.space.headless.apio.impl
+Bundle-Version: 2.0.0
```
Comment (nhpatt) [inline modules/apps/headless/content-space/content-space-headless-apio-impl/bnd.bnd]: Does it makes sense to have both packages now? We have to discuss it.

File: modules/apps/headless/content-space/content-space-headless-apio-impl/src/main/java/com/liferay/content/space/headless/apio/internal/architect/resource/ContentSpaceActionRouterImpl.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+/**
+ * Provides the information necessary to expose content space resources through
+ * a web API. The resources are mapped from the internal model {@code Group}.
+ *
+ * @author Javier Gamarra
+ * @author Cristina González
+ */
+@Component(immediate = true, service = ActionRouter.class)
+public class ContentSpaceActionRouterImpl
+	implements ContentSpaceActionRouter, ActionRouter<ContentSpace> {
```
Comment (nhpatt) [inline modules/apps/headless/content-space/content-space-headless-apio-impl/src/main/java/com/liferay/content/space/headless/apio/internal/architect/resource/ContentSpaceActionRouterImpl.java]: Although it's going to be generated I think that ContentSpaceActionRouter should extend ActionRouter<ContentSpace>

File: modules/apps/headless/content-space/content-space-headless-apio-api/bnd.bnd (hunk 1 above)
Comment (dgarciasarai) [reply modules/apps/headless/content-space/content-space-headless-apio-api/bnd.bnd]: This is fixed in the other pull request 👍

File: modules/apps/headless/content-space/content-space-headless-apio-impl/bnd.bnd (hunk 5 above)
Comment (ruben-pulido) [reply modules/apps/headless/content-space/content-space-headless-apio-impl/bnd.bnd]: We'll go with two modules as discussed

**Pull request closed without merging**